doc/examples/joystick/lib/player.dart

```diff
@@ -0,0 +1,91 @@
+import 'dart:math';
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+import 'package:flame/components/joystick/joystick_component.dart';
+import 'package:flame/components/joystick/joystick_events.dart';
+import 'package:flame/palette.dart';
+
+class Player extends Component implements JoystickListener {
+  final _whitePaint = BasicPalette.white.paint;
+  final _bluePaint = Paint()..color = const Color(0xFF0000FF);
+  final _greenPaint = Paint()..color = const Color(0xFF00FF00);
+  final double speed = 159;
+  double currentSpeed = 0;
+  double _radAngle;
+  bool _move = false;
+  Paint _paint;
+
+  Rect _rect;
+
+  Player() {
+    _paint = _whitePaint;
+  }
+
+  @override
+  void render(Canvas canvas) {
+    if (_rect != null) {
+      canvas.save();
+      canvas.translate(_rect.center.dx, _rect.center.dy);
+      canvas.rotate(_radAngle == 0.0 ? 0.0 : _radAngle + (pi / 2));
+      canvas.translate(-_rect.center.dx, -_rect.center.dy);
+      canvas.drawRect(_rect, _paint);
+      canvas.restore();
+    }
+  }
+
+  @override
+  void update(double dt) {
+    if (_move) {
+      moveFromAngle(dt);
+    }
+  }
+
+  @override
+  void resize(Size size) {
+    _rect = Rect.fromLTWH(
+      (size.width / 2) - 25,
+      (size.height / 2) - 25,
+      50,
+      50,
+    );
+    super.resize(size);
+  }
+
+  @override
+  void joystickAction(JoystickActionEvent event) {
+    if (event.event == ActionEvent.DOWN) {
+      if (event.id == 1) {
+        _paint = _paint == _whitePaint ? _bluePaint : _whitePaint;
+      }
+      if (event.id == 2) {
+        _paint = _paint == _whitePaint ? _greenPaint : _whitePaint;
+      }
+    }
+  }
+
+  @override
+  void joystickChangeDirectional(JoystickDirectionalEvent event) {
+    _move = event.directional != JoystickMoveDirectional.IDLE;
+    if (_move) {
+      _radAngle = event.radAngle;
+      currentSpeed = speed * event.intensity;
+    }
```

Comment:
if it's idle don't you need to reset currentSpeed = 0?

Reply:
It doesn't go to 0. but the 'move' is false, so don't move. but I could leave that out of the if without problems.

Reply:
no, I gotcha, my bad. didn't understood the _move field before.